src/modules/identity/RadixIdentity.ts

```diff
@@ -1,13 +1,19 @@
 import RadixSignatureProvider from './RadixSignatureProvider'
 import RadixDecryptionProvider from './RadixDecryptionProvider'
 import RadixAccount from '../account/RadixAccount'
+import RadixKeyPair from '../wallet/RadixKeyPair'
 
 import { RadixAtom } from '../RadixAtomModel'
 
 export default abstract class RadixIdentity implements RadixSignatureProvider, RadixDecryptionProvider {
+    constructor(readonly keyPair: RadixKeyPair) {
+        this.account = new RadixAccount(keyPair)
+        this.account.enableDecryption(this)
+    }
+
     public abstract signAtom(atom: RadixAtom): Promise<RadixAtom>
     public abstract decryptECIESPayload(payload: Buffer): Promise<Buffer>
-    public abstract getPublicKey(): Buffer
+    public abstract getPublicKey(): Promise<Buffer>
```

Comment:
Hmm, I'm not a fan of making this a promise - the public key cannot change, so you should only have to get it once when setting up a connection. Promises make everything else a lot more complicated

Reply:
I agree, but the problem here is that in order to get the pubic key the first time I need to make a request through a web socket and we don't know how long it'll take so we might have to wait for the response.

Reply:
But you have a private method getRemotePublicKey for that, no?

Reply:
You're right, for some reason I miss that, I'll make it a non-promise once again.